context/app/static/js/pages/Dataset/hooks.ts

```diff
@@ -177,13 +177,16 @@ function useProcessedDatasetsSections(): { sections: TableOfContentsItem | false
 }
 
 export function useRedirectAlert() {
-  const { redirected } = useFlaskDataContext();
+  const { redirected, redirectedFromId, redirectedFromPipeline } = useFlaskDataContext();
   const { toastInfo } = useSnackbarActions();
+
   useEffect(() => {
     if (redirected) {
-      toastInfo('You have been redirected to the unified view for this dataset.');
+      toastInfo(
+        `You have been redirected to the unified view for ${redirectedFromPipeline} dataset ${redirectedFromId}.`,
```

Comment:
If `redirectedFromId` and `redirectedFromPipeline` are both potentially undefined you should account for that possibility here.